### PR TITLE
Feature/template merge

### PR DESCRIPTION
**Pull request by Zethson**
Failing locally for now. See Slack for Details.

### CODE REVIEW

**Comment by Zethson** [comment]
So this PR is still using the old container.

**Comment by Zethson** [comment]
dev build is only triggered when it's merged to dev, right?

**Comment by Zethson** [comment]
@Leon-Bichmann To summarize:

1. We had conflicts with the new markdown stuff introduced in the template in our environment.
-> A few dependency upgrades solved that. It's likely mhcnuggets or percolator, but I did not investigate further which one actually solved it.

2. An update from OpenMS 2.4.0 to 2.5.0 breaks MHCquant in a very weird way. Only 3 out of 4 database searches complete successfully. I do not know why this is happening. May be required to dig into the OpenMS changelog or ask @timosachsenberg .

3. Please pull this branch, build the container locally and test that it runs through. It does now for me.

Then we can merge this to dev and the real dev container will be build and pushed to Dockerhub.

File: docs/output.md [hunk 1]
```diff
@@ -18,9 +18,9 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
 
 ## mzTab
 
-[mzTab](http://www.psidev.info/mztab) is a light-weight format to report mass spectrometry search results. It provides all important information about idenfied peptide hits and is compatible with the PRIDE Archive - proteomics data repository:
+[mzTab](http://www.psidev.info/mztab) is a light-weight format to report mass spectrometry search results. It provides all important information about identified peptide hits and is compatible with the PRIDE Archive - proteomics data repository:
 
-Griss, J. et al. The mzTab Data Exchange Format: Communicating Mass-spectrometry-based Proteomics and Metabolomics Experimental Results to a Wider Audience. Mol Cell Proteomics 13, 2765–2775 (2014).
+Griss, J. et al. The mzTab Data Exchange Format: Communicating Mass-spectrometry-based Proteomics and Metabolomics Experimental Results to a Wider Audience. Mol Cell Proteomics 13, 2765-2775 (2014)
```
Comment (Zethson) [inline docs/output.md]: This actually broke the new markdown script. I assume it was some copypasta that introduced a non utf 8 hyphen.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,5 @@
 # nf-core/mhcquant: Changelog
 
-The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
-and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
-
```
Comment (drpatelh) [inline CHANGELOG.md]: May want to put a `## dev` header in here ([see](https://github.com/nf-core/atacseq/blob/dev/CHANGELOG.md)) for changes made in the `dev` version. Also, worth adding a note that you have updated to tools 1.9

File: main.nf [hunk 3]
```diff
@@ -20,51 +20,47 @@ def helpMessage() {
...
-      --max_rt_alignment_shift          Maximal retention time shift (sec) resulting from linear alignment      
-      --spectrum_batch_size             Size of Spectrum batch for Comet processing (Decrease/Increase depending on Memory Availability)
-      --description_correct_features    Description of correct features for Percolator (0, 1, 2, 4, 8, see Percolator retention time and calibration) 
-      --klammer                         Retention time features are calculated as in Klammer et al. instead of with Elude.
-      --predict_RT                      Retention time prediction for identified peptides
-      --skip_decoy_generation           Use a fasta databse that already includes decoy sequences
-      --quantification_fdr              Assess and assign ids matched between runs with an additional quantification FDR
-      --quantification_min_prob         Specify a minimum probability cut off for quantification
-
+      --peptide_min_length                Minimum peptide length for filtering
```
Comment (drpatelh) [inline main.nf]:
```suggestion
      --peptide_min_length [int]           Minimum peptide length for filtering
```

Comment (drpatelh) [reply main.nf]: Add variable type for all other parameters in this listing too.

File: main.nf [hunk 4]
```diff
@@ -20,51 +20,47 @@ def helpMessage() {
...
+      --max_rt_alignment_shift            Maximal retention time shift (sec) resulting from linear alignment      
+      --spectrum_batch_size               Size of Spectrum batch for Comet processing (Decrease/Increase depending on Memory Availability)
+      --description_correct_features      Description of correct features for Percolator (0, 1, 2, 4, 8, see Percolator retention time and calibration) 
+      --klammer                           Retention time features are calculated as in Klammer et al. instead of with Elude.
+      --predict_RT                        Retention time prediction for identified peptides
+      --skip_decoy_generation             Use a fasta databse that already includes decoy sequences
+      --quantification_fdr                Assess and assign ids matched between runs with an additional quantification FDR
+      --quantification_min_prob           Specify a minimum probability cut off for quantification
     Binding Predictions:	
         --predict_class_1                 Whether a class 1 affinity prediction using MHCFlurry should be run on the results - check if alleles are supported (true, false)
```
Comment (drpatelh) [inline main.nf]: Indents are off here.

Comment (drpatelh) [reply main.nf]: Need to add spaces between sections too.

File: main.nf [hunk 5]
```diff
@@ -302,16 +270,34 @@ if( params.include_proteins_from_vcf){
...
+ch_output_docs = file("$baseDir/docs/output.md", checkIfExists: true)
+
+// Has the run name been specified by the user?
+//  this has the bonus effect of catching both -name and --name
+custom_runName = params.name
+if (!(workflow.runName ==~ /[a-z]+_[a-z]+/)) {
+    custom_runName = workflow.runName
+}
+
+if (workflow.profile.contains('awsbatch')) {
```
Comment (drpatelh) [inline main.nf]: This section looks incomplete relative to the pipeline template.

File: main.nf [hunk 6]
```diff
@@ -302,16 +270,34 @@ if( params.include_proteins_from_vcf){
...
+    // Check outdir paths to be S3 buckets if running on AWSBatch
+}
+
 // Header log info
 log.info nfcoreHeader()
 def summary = [:]
+if (workflow.revision) summary['Pipeline Release'] = workflow.revision
+summary['Run Name']         = custom_runName ?: workflow.runName
 summary['Pipeline Name']  = 'nf-core/mhcquant'
 summary['Pipeline Version'] = workflow.manifest.version
```
Comment (drpatelh) [inline main.nf]: Indents are off here in this section. Easier to visualise if you can try and line up the `=`

File: main.nf [hunk 7]
```diff
@@ -398,16 +383,18 @@ process get_software_versions {
     percolator -h &> v_percolator.txt
     comet -p
     mhcflurry-predict --version &> v_mhcflurry.txt
+    echo 2.3.1 > v_mhcnuggets.txt
```
[the remark this answers is not in the document]
Comment (Zethson) [reply main.nf]: They're not even executable packages. So the version cannot even be extracted from error messages or something.

We're importing them into python scripts in /bin

**Comment by Zethson** [comment]
@drpatelh Great review, thanks.
Will implemented the changes soon.

Final verdict by @Leon-Bichmann as soon as he can take a look.

File: conf/base.config [hunk 8]
```diff
@@ -11,7 +11,6 @@
 
 process {
 
-  // TODO nf-core: Check the defaults for all processes
   cpus = { check_max( 1 * task.attempt, 'cpus' ) }
   memory = { check_max( 7.GB * task.attempt, 'memory' ) }
```
Comment (Leon-Bichmann) [inline conf/base.config]: Maybe in this or another PR we should raise the max memory and max run times

to 16GB and 24h at least for the processes comet_db_search and predict_neoeptiopes_from_vcf

Comment (Leon-Bichmann) [reply conf/base.config]: Issue: #94

File: CHANGELOG.md (hunk 2 above)
Comment (Zethson) [reply CHANGELOG.md]: https://github.com/Zethson/MHCquant/commit/054a68542f25314188e860dc26106f350f771000

File: README.md [hunk 9]
```diff
@@ -43,16 +43,15 @@ See [usage docs](docs/usage.md) for all of the available options when running th
...
-3. Pipeline HPC Cluster configuration
+2. Pipeline configuration
     * [Local installation](https://nf-co.re/usage/local_installation)
     * [Adding your own system config](https://nf-co.re/usage/adding_own_config)
+3. [Running the pipeline](docs/usage.md)
 4. [Output and how to interpret the results](docs/output.md)
 5. [Troubleshooting](https://nf-co.re/usage/troubleshooting)
-6. [Pipeline-specific Troubleshooting](docs/troubleshooting.md)
 
 ## Credits
```
[the remark this answers is not in the document]
Comment (Zethson) [reply README.md]: https://github.com/Zethson/MHCquant/commit/9144e7b5b8cc4ac0a8bb2f9a6f167949f5b8851b

File: docs/usage.md [hunk 10]
```diff
@@ -8,8 +8,8 @@
   * [Updating the pipeline](#updating-the-pipeline)
   * [Reproducibility](#reproducibility)
 * [Main arguments](#main-arguments)
+  * [`-profile`](#-profile)
   * [`--mzmls`](#--mzmls)
-  * [`--fasta`](#--fasta)
   * [`-profile`](#-profile-single-dash)
     * [`docker`](#docker)
```
[the remark this answers is not in the document]
Comment (Zethson) [reply docs/usage.md]: https://github.com/Zethson/MHCquant/commit/9cdede56d74bb7a8593e7edd8ef61c102eac79c6

File: nextflow.config [hunk 11]
```diff
@@ -154,7 +154,7 @@ manifest {
   description = 'Identify and quantify peptides from mass spectrometry raw data'
   mainScript = 'main.nf'
   nextflowVersion = '>=19.10.0'
-  version = '1.3'
+  version = 'dev'
```
[the remark this answers is not in the document]
Comment (Zethson) [reply nextflow.config]: https://github.com/Zethson/MHCquant/commit/13381b324e3800a09a6a89af107398ad94df0bf0

File: environment.yml [hunk 12]
```diff
@@ -1,13 +1,18 @@
...
+  - conda-forge::pygments=2.5.2
+  - conda-forge::r-rmarkdown=2.1
   - openms=2.4.0
   - comet-ms=2016013
   - python=2.7.15
-  - mhcflurry=1.2.2
-  - fred2=2.0.3
-  - mhcnuggets=2.3.1
\ No newline at end of file
+  - mhcflurry=1.4.3
```
[the remark this answers is not in the document]
Comment (Zethson) [reply environment.yml]: https://github.com/Zethson/MHCquant/commit/13381b324e3800a09a6a89af107398ad94df0bf0

File: main.nf (hunk 3 above)
Comment (Zethson) [reply main.nf]: https://github.com/Zethson/MHCquant/commit/e67b7116a4e68c45efb05323b2ebb31aa5908d1f

File: main.nf (hunk 4 above)
Comment (Zethson) [reply main.nf]: https://github.com/Zethson/MHCquant/commit/ea5eb696f5b4db358c039e1a1feed0bcee052b63

Comment (Zethson) [reply main.nf]: https://github.com/Zethson/MHCquant/commit/3fea36ec6fe378a9b1b740e4ea2bfbec987663e1

File: main.nf (hunk 5 above)
Comment (Zethson) [reply main.nf]: https://github.com/Zethson/MHCquant/commit/ed4d2f780601656b0c440f6b03c63dc024db4c05

Great find!

File: main.nf (hunk 6 above)
Comment (Zethson) [reply main.nf]: https://github.com/Zethson/MHCquant/commit/51c5b888b894f1082ba1a08fdb0f80e608c49860

File: conf/base.config (hunk 8 above)
Comment (Zethson) [reply conf/base.config]: https://github.com/Zethson/MHCquant/commit/8196505bb518d9e03bed0620df40783e69850d10 and https://github.com/Zethson/MHCquant/commit/154197eb06732c3f9b8c708cea32bc5ef2abfe81

**Comment by Zethson** [comment]
@drpatelh Implemented your requested changes, hope I didn't miss anything.

@Leon-Bichmann process high mem & time labels included #94 

Still runs for me locally. Someone pull the trigger haha :)

**Comment by Leon-Bichmann** [comment]
So why did the tests not pass?

**Comment by Zethson** [comment]
As explained above  they do not pass because the new dev container has not been build yet. This is expected.

Please test it locally.

**Comment by Leon-Bichmann** [comment]
I see ok

**Pull request merged**